File: cep-java-catcher/src/main/java/org/apache/flink/cep/event/PatternWrapper.java
package org.apache.flink.cep.event;

import org.apache.flink.cep.pattern.Pattern;

import java.util.ArrayList;
import java.util.List;

public class PatternWrapper implements CatcherEvent {

    private String id;

    private Pattern pattern;

    private List<String> rules = new ArrayList<>();

    public PatternWrapper(String id, Pattern pattern) {
        this.id = id;
        this.pattern = pattern;
    }

    public void setRules(List<String> rules) {
        this.rules = rules;
    }

    public List<String> getRules() {
        return this.rules;
    }

    public String getId() {
        return id;
    }

    public Pattern getPattern() {
        return pattern;
    }

    @Override
    public void close() throws Exception {}

    @Override
    public String toString() {
        return id;
    }
}

File: cep-scala-catcher/src/main/scala/org/apache/flink/cep/scala/model/PatternWrapper.scala
package org.apache.flink.cep.scala.model

import java.util

import org.apache.flink.cep.scala.pattern.Pattern
import org.apache.flink.cep.event

class PatternWrapper(patterId: String, pattern: Pattern, rules: Array[String]) {

  def asJava(): event.PatternWrapper = {
    val jPatternWrapper = new event.PatternWrapper(patterId, pattern.wrappedPattern)
    if (rules.length > 0) {
      jPatternWrapper.setRules(util.Arrays.asList(rules: _*))
    }
    jPatternWrapper
  }

}
